[refactor] Remove `test_quantizer.py` and reuse one of its test in `test_quantization.py`

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#27269 [refactor] Remove `test_quantizer.py` and reuse one of its test in `test_quantization.py`**
* #27239 [refactor] Factor out insertPrepackUnpackForLinear
* #27238 [refactor] InsertObservers
* #27237 [quant][graphmode][fix] Make inserted child module names unique

Summary:
Remove `test_quantizer.py`, add and rewrite one of the tests in `test_quantizer`
in `test_quantization.py`
The conv test is removed for now since conv pattern is still broken, we'll add another test
later

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D18182916](https://our.internmc.facebook.com/intern/diff/D18182916)

jerryzh168 committed:
[refactor] Move tests in test_quantizer.py to test_quantization.py

Summary:
att
Also enable the tests

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Move tests in test_quantizer.py to test_quantization.py"


Summary:
att
Also enable the tests

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Move tests in test_quantizer.py to test_quantization.py"


Summary:
att
Also enable the tests

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Move tests in test_quantizer.py to test_quantization.py"


Summary:
att
Also enable the tests

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Move tests in test_quantizer.py to test_quantization.py"


Summary:
att
Also enable the tests

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Move tests in test_quantizer.py to test_quantization.py"


Summary:
att
Also enable the tests

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Move tests in test_quantizer.py to test_quantization.py"


Summary:
att
Also enable the tests

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Move tests in test_quantizer.py to test_quantization.py"


Summary:
att
Also enable the tests

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Move tests in test_quantizer.py to test_quantization.py"


Summary:
att
Also enable the tests

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Move tests in test_quantizer.py to test_quantization.py"


Summary:
att
Also enable the tests

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

ZolotukhinM commented:
The added test looks differently from the two tests that were removed. Could you please explain in more details why this is equivalent (also, please add that to the commit message since right now it doesn't look like the test was moved)?

jerryzh168 commented:
for test_quantizer.py, the first test is removed, the second one is the one we move to test_quantization, and I also modified a bit to make it pass

jerryzh168 committed:
Update on "[refactor] Move tests in test_quantizer.py to test_quantization.py"


Summary:
att
Also enable the tests

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Move tests in test_quantizer.py to test_quantization.py"


Summary:
att
Also enable the tests

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Move tests in test_quantizer.py to test_quantization.py"


Summary:
att
Also enable the tests

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Move tests in test_quantizer.py to test_quantization.py"


Summary:
att
Also enable the tests

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Move tests in test_quantizer.py to test_quantization.py"


Summary:
att
Also enable the tests

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Move tests in test_quantizer.py to test_quantization.py"


Summary:
att
Also enable the tests

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Move tests in test_quantizer.py to test_quantization.py"


Summary:
att
Also enable the tests

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Move tests in test_quantizer.py to test_quantization.py"


Summary:
att
Also enable the tests

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Move tests in test_quantizer.py to test_quantization.py"


Summary:
att
Also enable the tests

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Move tests in test_quantizer.py to test_quantization.py"


Summary:
att
Also enable the tests

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Move tests in test_quantizer.py to test_quantization.py"


Summary:
att
Also enable the tests

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Move tests in test_quantizer.py to test_quantization.py"


Summary:
att
Also enable the tests

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Move tests in test_quantizer.py to test_quantization.py"


Summary:
att
Also enable the tests

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

ZolotukhinM commented on test/test_quantization.py:
```diff
@@ -695,10 +696,49 @@ def test_single_layer(self):
         result_eager = model_eager(self.calib_data[0][0])
-        torch._C._jit_pass_quant_fusion(model_script._c._get_module('fc1')._get_method('forward').graph)
-        result_script = model_script._c._get_method('forward')(self.calib_data[0][0])
+        result_script = get_forward(model_script._c)(self.calib_data[0][0])
```
Why did this change if that was just a move of the test?

jerryzh168 replied:
this is move + some modifications to make it pass I'll modify the message

ZolotukhinM commented on test/test_quantization.py:
```diff
@@ -695,10 +696,49 @@ def test_single_layer(self):
         self.assertEqual(result_eager, result_script)
 
+    @unittest.skip("quantization for inlined linear is not working right now")
+    def test_nested(self):
```
It seems that we had more tests before the move, could you please explain why they were changed?

jerryzh168 replied:
the conv+bn test is added later after we fix the conv patterns

jerryzh168 committed:
Update on "[refactor] Move tests in test_quantizer.py to test_quantization.py"


Summary:
att
Also enable the tests

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Remove `test_quantizer.py` and reuse one of its test in `test_quantization.py`"


Summary:
Remove `test_quantizer.py`, add and rewrite one of the tests in `test_quantizer`
in `test_quantization.py`
The conv test is removed for now since conv pattern is still broken, we'll add another test
later

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Remove `test_quantizer.py` and reuse one of its test in `test_quantization.py`"


Summary:
Remove `test_quantizer.py`, add and rewrite one of the tests in `test_quantizer`
in `test_quantization.py`
The conv test is removed for now since conv pattern is still broken, we'll add another test
later

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D18182916](https://our.internmc.facebook.com/intern/diff/D18182916)

[ghstack-poisoned]

ZolotukhinM reviewed (approved):
Looks good! In future when you move stuff around please try to split it from non-mechanical changes for easier reviews.

jerryzh168 committed:
Update on "[refactor] Remove `test_quantizer.py` and reuse one of its test in `test_quantization.py`"


Summary:
Remove `test_quantizer.py`, add and rewrite one of the tests in `test_quantizer`
in `test_quantization.py`
The conv test is removed for now since conv pattern is still broken, we'll add another test
later

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D18182916](https://our.internmc.facebook.com/intern/diff/D18182916)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Remove `test_quantizer.py` and reuse one of its test in `test_quantization.py`"


Summary:
Remove `test_quantizer.py`, add and rewrite one of the tests in `test_quantizer`
in `test_quantization.py`
The conv test is removed for now since conv pattern is still broken, we'll add another test
later

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D18182916](https://our.internmc.facebook.com/intern/diff/D18182916)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Remove `test_quantizer.py` and reuse one of its test in `test_quantization.py`"


Summary:
Remove `test_quantizer.py`, add and rewrite one of the tests in `test_quantizer`
in `test_quantization.py`
The conv test is removed for now since conv pattern is still broken, we'll add another test
later

Test Plan:
python test/test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D18182916](https://our.internmc.facebook.com/intern/diff/D18182916)

[ghstack-poisoned]
